lib/ClangImporter/ClangImporter.cpp

```diff
@@ -1686,11 +1686,23 @@ bool ClangImporter::Implementation::importHeader(
 
   // We can't do this as we're parsing because we may want to resolve naming
   // conflicts between the things we've parsed.
-  for (auto group : allParsedDecls)
-    for (auto *D : group)
-      if (auto named = dyn_cast<clang::NamedDecl>(D))
+  for (auto group : allParsedDecls) {
+    for (auto *D : group) {
+      // Iterate into extern "C" {} type declarations.
+      if (auto linkageDecl = dyn_cast<clang::LinkageSpecDecl>(D)) {
```

Comment:
Technically, `extern` blocks can be nested, so I think this should be something like:
```cpp
  std::function<void(clang::Decl *)> visit;
  visit = [&](clang::Decl *d) {
    // Iterate into extern "C" {} type declarations.
    if (auto linkageDecl = dyn_cast<clang::LinkageSpecDecl>(D)) {
      for (auto *decl : linkageDecl->noload_decls()) {
        visit(decl);
      }
    }
    if (auto named = dyn_cast<clang::NamedDecl>(D)) {
      addEntryToLookupTable(*BridgingHeaderLookupTable, named,
                              getNameImporter());
    }
  };
  
  for (auto group : allParsedDecls) {
    for (auto *D : group) {
      visit(D);
    }
  }
```
But nobody uses nested `extern`, so 🤷